app/js/ProjectPersistenceManager.js

```diff
@@ -61,26 +63,39 @@ module.exports = ProjectPersistenceManager = {
   },
 
   refreshExpiryTimeout: callbackify(refreshExpiryTimeout),
-  markProjectAsJustAccessed(project_id, callback) {
-    if (callback == null) {
-      callback = function (error) {}
-    }
-    const timer = new Metrics.Timer('db-bump-last-accessed')
-    const job = (cb) =>
-      db.Project.findOrCreate({ where: { project_id } })
-        .spread((project, created) =>
-          project
-            .update({ lastAccessed: new Date() })
-            .then(() => cb())
-            .error(cb)
+
+  init() {
+    fs.readdir(Settings.path.compilesDir, (err, dirs) => {
+      if (err) {
+        logger.warn({ err }, 'cannot get project listing')
+        return
+      }
+
+      async.eachLimit(dirs, 50, (projectAndUserId, cb) => {
```

Comment:
Looking at this again, and given that we've increased the threads to 128, maybe a parallelism of 50 is too high?

Perhaps we should move the setInterval at [app.js#L415-L419](https://github.com/overleaf/clsi/blob/fc87a0e40ce62055066ccfbaf38c912386800146/app.js#L415-L419) into the `init` method so we only start expiring projects after the initialisation is complete, and then we don't need to worry about how quickly we scan the directory.

Reply:
Good ideas, done in e9af3a4d033ec4d8fc8ebcff3337a20b3e2e62c6